### DISCUSSION TITLE
Starting simulation

### DISCUSSION THREAD

**Discussion by mikhailpaltsev** [OpenGATE/opengate#543] [Technical issues] [answered]
After the code is written, how do I properly run its simulation and view the results and visualization? Is it possible to run a simulation in visual studio? 
When I run the test code in Visual studio, it appears ". ModuleNotFoundError: No module named 'opengate' ". Although it was performed during the tests. I am sure that I am approaching the simulation incorrectly.
Thanks!

**Comment by guneet19** [comment]
Hello,

You can run the code directly from Visual Studio Code. Did you create a conda environment while installing? If yes, make sure that you change your kernel in VS Code to that environment before running your simulation files.

**Reply by mikhailpaltsev** [reply]
No, I did not install the Conda. Does it need to be installed to run?

**Reply by tbaudier** [reply]
Hello,

The installation procedure is described here: https://github.com/OpenGATE/opengate?tab=readme-ov-file#how-to-install-short-version with venv or conda (choose one of them).
Once installed, configure VS Code choosing the python of your environement.

**Reply by mikhailpaltsev** [reply]
> Hello,
> 
> The installation procedure is described here: https://github.com/OpenGATE/opengate?tab=readme-ov-file#how-to-install-short-version with venv or conda (choose one of them). Once installed, configure VS Code choosing the python of your environement.

Hello, what exactly should I choose in opengate_env as an interpreter?

**Reply by mikhailpaltsev** [reply]
> Hello,
> 
> The installation procedure is described here: https://github.com/OpenGATE/opengate?tab=readme-ov-file#how-to-install-short-version with venv or conda (choose one of them). Once installed, configure VS Code choosing the python of your environement.
"No module named opengate" 

Maybe I'm misplacing the files I want to run? In my opengate_env, I created a copy of the test to try to run the simulation myself. I open it in VS and select it as an interpreter (I haven't figured out what yet) and click start debugging...

When choosing my opengate _env as the folder with the interpreter, I fall inside and do not understand what to choose next

**Reply by tbaudier** [reply]
You can find more information here: https://code.visualstudio.com/docs/python/environments#_manually-specify-an-interpreter

**Reply by mikhailpaltsev** [reply]
> You can find more information here: https://code.visualstudio.com/docs/python/environments#_manually-specify-an-interpreter

Thanks for the reply. But I'm completely confused about how to properly run my examples using VS. I hope the developers will pay attention to this question from those who are just starting their way in opengate10

**Reply by guneet19** [reply]
Can you try another way? Can you open your terminal, activate your environment, go to the folder where you have the simulation file, and run it using `python3 file_name` ? If it still throws an error, can you attach an image of the error?

**Reply by mikhailpaltsev** [reply]
> Can you try another way? Can you open your terminal, activate your environment, go to the folder where you have the simulation file, and run it using `python3 file_name` ? If it still throws an error, can you attach an image of the error?

![Снимок экрана 2024-11-13 в 16 07 58](https://github.com/user-attachments/assets/606f187a-9413-495b-ab61-7fc32e5afb5e)

**Reply by guneet19** [reply, marked as answer]
Hello,

I just did a reinstall with a python venv as shown in the documentation and it worked for me without an error.  
![image](https://github.com/user-attachments/assets/9e24f4c5-152d-4379-8ab9-279681f043e7)

Steps I followed - 

1 - create a virtual environment and activate it - 

`python -m venv opengate_env`
`source opengate_env/bin/activate`

2 - Install opengate 

`pip install --upgrade pip`
`pip install --pre opengate`

3 - Run tests

`opengate_tests` When you do this step required Geant4 data and test data will be downloaded. 

4 - Downloaded test006_runs.py

5 - Run it with python3 test006_runs.py

Can you try reinstalling by following these steps...

**Reply by mikhailpaltsev** [reply]
Thanks! I'll try to reinstall it, but I remembered what could have prevented it

I had python 3.13 installed, but during installation I reinstalled it because only version 3.12 is compatible. 

By the way, all my tests were tested, I just wanted to run some of the tests separately as an example of modeling as training

**Reply by mikhailpaltsev** [reply]
Thank you, the test has already started

But anyway, if you run it through VS, it doesn't find opengate.

'    import opengate as gate
ModuleNotFoundError: No module named 'opengate'"
I changed "opengate_env" to "opengate" and it began to successfully perform the import opengate action. But it did not find the module opengate.tests. Of course, because there is no tests folder in this environment. There is only bin, lib and so on. I came to the conclusion that my interpreter was incorrectly selected. In the folder env-lib-python3.12-sitepackage-opengate  there is a tests folder, maybe it is this opengate that he should find? If so, which interpreter is needed? And from which folder should I choose it?

**Comment by mikhailpaltsev** [comment]
The simulation can only be started using the terminal: python file.py ? Or you can use visual studio, but then it does not find the opengate module.  Apparently, in the VS code it is necessary to activate the source openagate_env/bin/activate somehow?

**Comment by dsarrut** [comment]
Hi, 
whatever the system (linux, windows, mac), you must activate the environment before executing the script. I dont have experience with VS but it should be feasible. Maybe other people with VS can help here ?
